application/libraries/DI.php

```diff
@@ -41,34 +42,49 @@ public static function getContainer()
      */
     public static function makeContainer()
     {
-        $container = new \DI\Container;
+        $builder = new ContainerBuilder();
+        $builder->addDefinitions([
+            LSYii_Application::class => function () {
+                //exit('C');
+                return App();
+            },
+            PluginManager::class => function () {
+                return App()->getPluginManager();
+            },
+            CHttpSession::class => function () {
+                return App()->session;
+            },
+            CDbConnection::class => function () {
+                return App()->db;
+            }
+        ]);
 
-        // Type hinting on a Yii model / active-record class should return its
-        // - static instance e.g Survey::model(). It is not possible to type hint
-        // - on this, so instead we configure the container to return the correct
-        // - object based on the call time class name, whenever we type hint on
-        // - CActiveRecord or anything that extends CActiveRecord
-        $container->set(CActiveRecord::class, function (CActiveRecord $entry) {
-            $class = $entry->getName();
-            return $class::model();
-        });
-
-        $container->set(LSYii_Application::class, function () {
-            return App();
-        });
-
-        $container->set(PluginManager::class, function () {
-            return App()->getPluginManager();
-        });
-
-        $container->set(CHttpSession::class, function () {
-            return App()->session;
-        });
-
-        $container->set(CDbConnection::class, function () {
-            return App()->db;
-        });
+        return $builder->build();
+    }
 
-        return $container;
+    private static function getActiveRecordDefinitions()
```

Comment:
never used?

Reply:
Turns out that the CActiveRecord DI doesn't work. The problem is that container->make() returns MyModel::model(), which is not what we want. We only want that for container->get().

Regardless, models are working correctly even without DI working as expected. Possibly at the cost of some performance due to multiple model instances being used for querying. I don't have a solution for this.

Lets consider this PR a refactor of the DI container. It doesn't solve CActiveRecord issue but its an improvement to the DI config.